porch/pkg/git/package.go

```diff
@@ -203,6 +203,10 @@ func (p *gitPackageRevision) GetUpstreamLock() (kptfile.Upstream, kptfile.Upstre
 	return *kf.Upstream, *kf.UpstreamLock, nil
 }
 
+func (p *gitPackageRevision) GetLock() (kptfile.Upstream, kptfile.UpstreamLock, error) {
```

Comment:
I suggest using this new `GetLock` to replace the lines here:

https://github.com/GoogleContainerTools/kpt/blob/919b09de3b8c3a6663557a997a604f2379ae9232/porch/pkg/engine/clone.go#L92-L100

Currently, it is clearing the Kptfile and then making a call to `GetUpstreamLock`, which reduces down to just `p.findParent()`.

If we change the clone task to use this instead of `GetUpstreamLock`, we can also change line 201 in this file to just return empty structs. That will simplify the code a bit - `GetUpstreamLock` will only look at the Kptfile and `GetLock` will only look at the parent.

Reply:
Following up, @ChristopherFry reported that he's finding that packages created from scratch are getting an upstreamLock pointing to itself. I think this is because the `GetUpstreamLock` method currently points to `parent` as a backup (which is wrong), and I think my suggestion here will fix the bug.

Reply:
Ack.

Reply:
Addressed this in the most recent commit.